packages/provider/src/RelayClient.ts

```diff
@@ -156,6 +163,11 @@ export class RelayClient {
     if (!this.config.skipErc165Check) {
       await this.dependencies.contractInteractor._validateERC165InterfacesClient()
     }
+
+    const keypair = nacl.box.keyPair()
+    this.publicKey = hexlify(keypair.publicKey)
+    const sharedKey = await this.dependencies.contractInteractor.getHubSymmetricKey(this.publicKey)
+    this.cipher = new deoxysii.AEAD(arrayify(sharedKey))
```

Comment:
Please remember that this will be replaced with `X25519DeoxysII` from the compat lib!